corehq/apps/app_manager/add_ons.py

```diff
@@ -0,0 +1,264 @@
+        description=_("Choose whether each form in a case list should register, update, or close a case."),
+        help_link="https://confluence.dimagi.com/display/commcarepublic/Case+Configuration",
+    ),
+    "enum_image": AddOn(
+        name=feature_previews.ENUM_IMAGE.label,
+        description=feature_previews.ENUM_IMAGE.description,
+        help_link=feature_previews.ENUM_IMAGE.help_link,
+        used_in_module=lambda m: _uses_detail_format(m, 'enum-image'),
+    ),
+    "menu_mode": AddOn(
+        name=_("Menu Mode"),
+        description=_("Control whether a form's enclosing menu is displayed on the mobile device or not. "
+        "Available in menu settings."),
+        used_in_module=lambda m: getattr(m, 'put_in_root', False),
+    ),
+    "register_from_case_list": AddOn(
+        name=_("Register from case list"),
+        description=_("Minimize duplicates by making registration forms available directly from the case list "
+        "on the mobile device. Availabe in menu settings."),
+        help_link="https://confluence.dimagi.com/pages/viewpage.action?pageId=30605985",
+        used_in_module=lambda m: m.case_list_form.form_id,
+    ),
+    "subcases": AddOn(
+        name=_("Child Cases"),
+        description=_("Open other types of cases, linking them to the case that "
+        "created them. Available in form settings."),
+        help_link="https://confluence.dimagi.com/display/commcarepublic/Child+Cases",
+        used_in_form=lambda f: f.form_type != "module_form" or bool(f.actions.subcases),
+    ),
+    "empty_case_lists": AddOn(
+        name=_("New Case Lists Created Empty"),
+        description=_("When adding a new case list, don't include a registration and followup form."),
+    ),
+}
+
+_LAYOUT = [
+    {
+        "slug": "case_management",
+        "collapse": False,
+        "name": _("Case Management"),
+        "description": _("Build more complex workflows"),
+        "slugs": ["conditional_form_actions", "edit_form_actions", "empty_case_lists", "subcases"],
+    },
+    {
+        "slug": "mobile",
+        "collapse": True,
+        "name": _("Mobile Experience"),
+        "description": _("Improve the user experience of your mobile workers"),
+        "slugs": ["case_list_menu_item", "enum_image", "menu_mode", "register_from_case_list"],
+    },
+    {
+        "slug": "xpath",
+        "collapse": True,
+        "name": _("Calculations"),
+        "description": _("Add logic to your app with XPath expressions"),
+        "slugs": ["display_conditions", "conditional_enum", "calc_xpaths", "advanced_itemsets"],
+    },
+    {
+        "slug": "efficiency",
+        "collapse": True,
+        "name": _("App Building Efficiency"),
+        "description": _("Tools to help build your apps faster"),
+        "slugs": ["case_detail_overwrite"],
+    },
+]
+
+
+# Determine whether or not UI should show a feature, based on
+# availability and whether or not it's in use.
+@memoized
```

Comment:
Also since the request is an arg this will only ever hit the cache for calls within the same request.